reactioncommerce/reaction-component-library
# (feat): 160 nnnnat shipping address checkout action

Resolves #160 #113 
Impact: **minor**  
Type: **feature**

## Components
Created the `CheckoutActions` component based on these [specs](https://github.com/reactioncommerce/reaction-component-library/issues/113)
The `CheckoutActions` component is responsible for:
  * Displaying each `CheckoutAction` and managing it's status. (complete, incomplete, active)
  * Providing each  `CheckoutAction` with any data it may need from the `cart` object.
  * Capturing/Editing of `CheckoutAction` data.
  * Rendering captured `CheckoutAction` data in a `CheckoutActionComplete` component.

Created the `ShippingAddressCheckoutAction` component based on these [specs](https://github.com/reactioncommerce/reaction-component-library/issues/160).
The `ShippingAddressCheckoutAction` serves 4 purposes:
  * To normalize data input/editing between the `CheckoutActions` component and the `AddressForm` components.
  * To provide `CheckoutActions` with a `renderComplete` method that renders the captured checkout data.
  * To provide `ChecoutActions` a way to know the action is ready for save,
  * To Provide `CheckoutActions` a way to capture the `AddressForm` values.


## Notes
The `CheckoutActions` component still has some gray areas on how to best provide each "Checkout Action" component the correct data from the `cart` object.  This is something I think will be worked out during implementation.

## Testing
* Verify the `CheckoutActions` component is working
    *  Add a new shipping address. (Since the example is using a mock mutation call it will return a mock address via props once the mutation resolves)
    * Verify the action will display as complete when provided `cartData`


* Verify `ShippingAddressCheckoutAction` is working.
  * Verify the `onReadyForChange` callback is firing when the form is filled.
  * Verify the `renderComplete` instance method is working correctly.
